File: Go/uno/pkg/AddTwoNumbers/AddTwoNumbers.go
package AddTwoNumbers

func addTwoNumbers(l1 *ListNode, l2 *ListNode) *ListNode {
	var head, tail *ListNode
	var carry int
	for l1 != nil || l2 != nil || carry != 0 {
		var sum int
		if l1 != nil {
			sum += l1.Val
		}
		if l2 != nil {
			sum += l2.Val
		}
		sum += carry
		carry = sum / 10
		sum = sum % 10
		node := &ListNode{Val: sum}
		if head == nil {
			head = node
			tail = node
		} else {
			tail.Next = node
			tail = node
		}
		if l1 != nil {
			l1 = l1.Next
		}
		if l2 != nil {
			l2 = l2.Next
		}
	} // for>
	return head
}
func addTwoNumbers2(l1 *ListNode, l2 *ListNode) *ListNode {
	stepOn := 0
	var result = ListNode{}
	var head = &result

	for {
		if l1 == nil || l2 == nil {
			break
		}
		sum := l1.Val + l2.Val + stepOn
		stepOn = 0
		if sum >= 10 {
			sum -= 10
			stepOn = 1
		}
		l1 = l1.Next
		l2 = l2.Next
		head.Val = sum
		if l1 != nil || l2 != nil || stepOn > 0 {
			head.Next = &ListNode{}
		}
		head = head.Next
	}

	rest := l1
	if rest == nil {
		rest = l2
	}
	for rest != nil {
		sum := rest.Val + stepOn
		stepOn = 0
		if sum >= 10 {
			sum -= 10
			stepOn = 1
		}
		rest = rest.Next
		head.Val = sum
		if rest != nil || stepOn > 0 {
			head.Next = &ListNode{}
		}
		head = head.Next
	}
	if head != nil {
		head.Val = stepOn

	}
	return &result
}

File: csharp/uno/AddTwoNumbers/AddTwoNumbers.cs
using Csharp.ListNode;

namespace Csharp.AddTwoNumbers
{
    public class AddTwoNumbers
    {
        public ListNode addTwoNumbers(ListNode l1, ListNode l2)
        {
            ListNode result = new ListNode(0);
            ListNode current = result;
            int carry = 0;
            while (l1 != null || l2 != null)
            {
                int x = (l1 != null) ? l1.val : 0;
                int y = (l2 != null) ? l2.val : 0;
                int sum = carry + x + y;
                carry = sum / 10;
                current.next = new ListNode(sum % 10);
                current = current.next;
                if (l1 != null) l1 = l1.next;
                if (l2 != null) l2 = l2.next;

            }
            if (carry > 0)
            {
                current.next = new ListNode(carry);
            }
            return result.next;
        }
        public ListNode AddTwoNumbers(ListNode l1, ListNode l2)
        {
            ListNode resultHead = new ListNode();
            ListNode resultCur = resultHead;
            while (l1 != null || l2 != null)
            {
                int n1 = (l1 != null) ? l1.val : 0;
                int n2 = (l2 != null) ? l2.val : 0;

                if (l1 != null) l1 = l1.next;
                if (l2 != null) l2 = l2.next;



                resultCur.val += n1 + n2;



                if (resultCur.val >= 10)
                {
                    resultCur.val -= 10;
                    resultCur.next = new ListNode(1);


                }
                if (((l1 != null) || (l2 != null)) && resultCur.next == null)
                {
                    resultCur.next = new ListNode();


                }


                resultCur = resultCur.next;

            }
            return resultHead;
        }
    }
}